### PR TITLE
Depend on nbconvert-core

### PR DESCRIPTION
**Pull request by chrisburr**
Similar to https://github.com/conda-forge/notebook-feedstock/pull/107, see https://github.com/conda-forge/jupyter-feedstock/pull/25#issuecomment-1303193663 for the motivation.

### CODE REVIEW

**Comment by chrisburr** [comment]
@conda-forge/nbclassic Ready for review, see the PR description for details.

**Comment by bollwyvl** [comment]
yeah, this would eventually come up. As a common "leaf" dependency, I think the confusion this would cause for end users of not having the pandoc-related features would be a confusing issue for folk, with the broader discussion here: https://github.com/conda-forge/nbconvert-feedstock/issues/72

Is there a way we can do a multiple `outputs` that would ensure pandoc for the "happy path" on pandoc-supporting subdirs?

**Comment by bollwyvl** [comment]
Indeed, if I was more confident in the PDF options, i'd want to ship `nbconvert-webpdf` in the `jupyter` MEGAMETAPACKAGE... but it's still pretty hairy.

**Comment by chrisburr** [comment]
I found myself debugging this same conflict again 🤦

> Is there a way we can do a multiple outputs that would ensure pandoc for the "happy path" on pandoc-supporting subdirs?

I think this trick should work nicely with modifing the build number, what do you think @bollwyvl?

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+use_pandoc:
+  - yes
```
Comment (bollwyvl) [inline recipe/conda_build_config.yaml:2]: i have fear of yaml turning `yes` and `no` into unexpected things...

Comment (bollwyvl) [reply recipe/conda_build_config.yaml:2]: maybe just

```yaml
with_nbconvert:
- nbconvert-core  # or "-core"
- nbconvert # or ""
```

and reuse them directly?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -33,7 +39,7 @@ requirements:
     - jupyter_client >=6.1.1
     - jupyter_core >=4.6.1
     - jupyter_server >=1.8
-    - nbconvert >=5
+    - nbconvert{{ "" if use_pandoc != "no" else "-core" }} >=5
```
Comment (bollwyvl) [inline recipe/meta.yaml:42]: probably need some kind of test that excites this so the failure mode is easy to identify...

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (chrisburr) [reply recipe/conda_build_config.yaml]: This is much more elegant!

> i have fear of yaml turning yes and no into unexpected things...

The fact my first (local) version failed due to be expecting `yes`/`true` to be turned into booleans proves this point 😆

File: recipe/meta.yaml (hunk 2 above)
Comment (chrisburr) [reply recipe/meta.yaml]: I don't think there is a practical way of testing this but I'm open to suggestions (I have tried it on a private channel to make sure it at works in it's current form at least).

Comment (bollwyvl) [reply recipe/meta.yaml]: real testing even better!

**Comment by bollwyvl** [comment]
maybe one thing missing (which would ideally land in the `meta.yaml`) is _why_ this is the case, maybe a link to a specifc issue for history... worried we'll forget as it does look a hair complicated.

**Review by bollwyvl — APPROVED** [review]

**Comment by bollwyvl** [comment]
So this built:

```
{
  "nbclassic-0.5.3-pyhb4ecaf3": {
    "recipe": {
      "channel_targets": "conda-forge main",
      "nbconvert_variant": "nbconvert"
    }
  }
}

```

and

```
{
  "nbclassic-0.5.3-pyh8b2e9e2": {
    "recipe": {
      "channel_targets": "conda-forge main",
      "nbconvert_variant": "nbconvert-core"
    }
  }
}
```

Which will the solver prefer?

**Comment by chrisburr** [comment]
It's easier to see from the file names which include the build number.

* `nbclassic-0.5.3-pyh8b2e9e2_2.conda` (`nbconvert_variant: nbconvert-core`)
* `nbclassic-0.5.3-pyhb4ecaf3_3.conda` (`nbconvert_variant: nbconvert`)

The latter is preffered due to the higher build number (the last digit of the hash is a coincidence).

Currently on the main channel it ends up favouring the Python 3.10 for `linux-ppc64le` build of Python because it can't use a newer one:

```bash
$ CONDA_SUBDIR=linux-64 mamba create --name test jupyterlab --dry-run 2>&1 | rg '(nbclassic| python )'
  + nbclassic                            0.5.2  pyhd8ed1ab_0        conda-forge/noarch          6MB
  + python                              3.11.0  he550d4f_1_cpython  conda-forge/linux-64       31MB
$ CONDA_SUBDIR=linux-aarch64 mamba create --name test jupyterlab --dry-run 2>&1 | rg '(nbclassic| python )'
  + nbclassic                            0.5.2  pyhd8ed1ab_0        conda-forge/noarch               6MB
  + python                              3.11.0  ha43d526_1_cpython  conda-forge/linux-aarch64       15MB
$ CONDA_SUBDIR=linux-ppc64le mamba create --name test jupyterlab --dry-run 2>&1 | rg '(nbclassic| python )'
  + nbclassic                            0.5.2  pyhd8ed1ab_0        conda-forge/noarch               6MB
  + python                              3.10.9  h062392f_0_cpython  conda-forge/linux-ppc64le       13MB
$ CONDA_SUBDIR=osx-64 mamba create --name test jupyterlab --dry-run 2>&1 | rg '(nbclassic| python )'
  + nbclassic                            0.5.2  pyhd8ed1ab_0        conda-forge/noarch        6MB
  + python                              3.11.0  he7542f4_1_cpython  conda-forge/osx-64       15MB
$ CONDA_SUBDIR=osx-arm64 mamba create --name test jupyterlab --dry-run 2>&1 | rg '(nbclassic| python )'
  + nbclassic                            0.5.2  pyhd8ed1ab_0        conda-forge/noarch           6MB
  + python                              3.11.0  h3ba56d0_1_cpython  conda-forge/osx-arm64       14MB
```

With this change in my local channel it favours `_3` from this PR but falls back to `_2`:

```bash
$ CONDA_SUBDIR=linux-64 mamba create -c local --name test jupyterlab --dry-run 2>&1 | rg '(nbclassic| python )'
  + nbclassic                            0.5.3  pyhb4ecaf3_3        /Users/cburr/mambaforge/conda-bld/noarch        8MB
  + python                              3.11.0  he550d4f_1_cpython  conda-forge/linux-64                           31MB
$ CONDA_SUBDIR=linux-aarch64 mamba create -c local --name test jupyterlab --dry-run 2>&1 | rg '(nbclassic| python )'
  + nbclassic                            0.5.3  pyhb4ecaf3_3        /Users/cburr/mambaforge/conda-bld/noarch        8MB
  + python                              3.11.0  ha43d526_1_cpython  conda-forge/linux-aarch64                      15MB
$ CONDA_SUBDIR=linux-ppc64le mamba create -c local --name test jupyterlab --dry-run 2>&1 | rg '(nbclassic| python )'
  + nbclassic                            0.5.3  pyh8b2e9e2_2        /Users/cburr/mambaforge/conda-bld/noarch        8MB
  + python                              3.11.0  h062392f_1_cpython  conda-forge/linux-ppc64le                      16MB
$ CONDA_SUBDIR=osx-64 mamba create -c local --name test jupyterlab --dry-run 2>&1 | rg '(nbclassic| python )'
  + nbclassic                            0.5.3  pyhb4ecaf3_3        /Users/cburr/mambaforge/conda-bld/noarch        8MB
  + python                              3.11.0  he7542f4_1_cpython  conda-forge/osx-64                             15MB
$ CONDA_SUBDIR=osx-arm64 mamba create -c local --name test jupyterlab --dry-run 2>&1 | rg '(nbclassic| python )'
  + nbclassic                            0.5.3  pyhb4ecaf3_3        /Users/cburr/mambaforge/conda-bld/noarch        8MB
  + python                              3.11.0  h3ba56d0_1_cpython  conda-forge/osx-arm64                          14MB
```

**Pull request merged**